File: heap-medium/1014.最佳观光组合.cpp
/*
 * @lc app=leetcode.cn id=1014 lang=cpp
 *
 * [1014] 最佳观光组合
 */
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution {
public:
    int maxScoreSightseeingPair(vector<int> &A) {
        int ret = 0, maxScore = A[0];
        for (int i = 1; i < A.size(); i++) {
            ret = max(ret, maxScore + A[i] - i);
            maxScore = max(maxScore, A[i] + i);
        }
        return ret;
    }
};
// @lc code=end

File: heap-medium/1014.最佳观光组合.py
#
# @lc app=leetcode.cn id=1014 lang=python3
#
# [1014] 最佳观光组合
#

# @lc code=start
class Solution:
    def maxScoreSightseeingPair(self, A: List[int]) -> int:
        ret, max_score = 0, A[0]
        for i in range(1, len(A)):
            ret = max(ret, max_score + A[i] - i)
            max_score = max(max_score, A[i] + i)
        return ret
# @lc code=end
